File: mowgli/lib/cskg/node.py
from typing import Optional, Tuple, Dict, NamedTuple


class Node(NamedTuple):
    id: str
    label: str
    aliases: Optional[Tuple[str, ...]] = None
    pos: Optional[str] = None
    # datasource is not optional, but it's ordered among the optional fields in the CSKG format
    datasource: str = ''
    other: Optional[Dict[str, object]] = None

File: gui/app/models/cskg/Node.scala
package models.cskg

final case class Node(
                       aliases: Option[List[String]], // No default values so the compiler can check missing fields on construction
                       datasource: String,
                       id: String,
                       label: String,
                       other: Option[String],
                       pos: Option[String],
                     )
